app/assets/javascripts/comfortable_mexican_sofa/admin/modules/CategoryExpand.js

```diff
@@ -0,0 +1,71 @@
+define(['jquery','DoughBaseComponent'], function ($, DoughBaseComponent) {
+'use strict';
+
+  var defaultConfig = {
+    categoriesCharacterCount: 20
+  };
+  var CategoryExpandProto;
+
+  function CategoryExpand($el, config) {
+    CategoryExpand.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  DoughBaseComponent.extend(CategoryExpand);
+
+  CategoryExpand.componentName = 'CategoryExpand';
+
+  CategoryExpandProto = CategoryExpand.prototype;
+
+  CategoryExpandProto.init = function(initialised) {
+
+    this.addExpanders();
+
+    $('#filter_form').on('ajax:complete', $.proxy(function() {
+      this.addExpanders();
+    }, this));
+
+    this._initialisedSuccess(initialised);
+      return this;
+    };
+
+    CategoryExpandProto.addExpanders = function(){
+
+    var self = this;
+
+    $('.js-filter-pages td:nth-child(3)').addClass('categories');
+
+    this.$el.find('td:nth-child(3)').each(function(){
+
+      var categories = $(this).text(),
+          categoryCount = categories.split(',').length - 1,
+          categoriesCharacterCount = self.config.categoriesCharacterCount;
+
+      if (categories.indexOf(',') >= 1) {
+        $(this).html(
+          categories.slice(0,categoriesCharacterCount)+
+          '<br /><a href="#" class="more">+' + categoryCount + ' see all</a>'+
+          '<span style="display:none;">'+ categories.slice(categoriesCharacterCount ,categories.length)+' <a href="#" class="less">Hide</a></span>'
+        );
+      };
+
+    });
+
+    this.$el.find('.more').click(function(event){
+      event.preventDefault();
+      $(this).hide().prev().hide();
+      $(this).next().show();
+    });
+
+    this.$el.find('.less').click(function(event){
+      event.preventDefault();
+      $(this).parent().hide().prev().show().prev().show();
```

Comment:
This seems a little fragile on first reading... do you think there's a way we could tighten it up so we know we're targetting the right thing?